primitives/src/crh/bowe_hopwood/mod.rs

```diff
@@ -161,6 +161,25 @@ impl<G: Group> Debug for BoweHopwoodPedersenParameters<G> {
     }
 }
 
+impl<G: Group> BoweHopwoodPedersenParameters<G>{
+    pub fn check_consistency(&self) -> bool {
```

Comment:
Why introducing this sanity check? If the setup is cannot be trusted (for example, by having produced the base points via hashing), there is no way to assure security: the linear relation between the base points might be any non-trivial, not detected by this function.